src/secrets/src/lib.rs

```diff
@@ -85,18 +85,14 @@ use mz_repr::GlobalId;
 
 /// Securely manages user secrets.
 #[async_trait]
-pub trait SecretsController: Debug + Send + Sync {
+pub trait SecretsController: SecretsReader {
     /// Creates or updates the specified secret with the specified binary
     /// contents.
     async fn ensure(&self, id: GlobalId, contents: &[u8]) -> Result<(), anyhow::Error>;
 
     /// Deletes the specified secret.
     async fn delete(&self, id: GlobalId) -> Result<(), anyhow::Error>;
 
-    /// Lists known secrets. Unrecognized secret objects do not produce an error
-    /// and are ignored.
-    async fn list(&self) -> Result<Vec<GlobalId>, anyhow::Error>;
-
```

Comment:
Having `list` be part of the `SecretsController` API was intentional. While it's technically a read-only operation, the idea was that the controller contains operations that are only relevant if you are managing secrets, and the `SecretsReader` contains the single operation (reading a secret) that is relevant if all you need is to determine the contents of secrets whose IDs you are told. 

I'm not seeing a good reason to change this! Keeping the `SecretsReader` as simple as possible makes it easier to do something like a `CachingSecretsReader` as discussed here: https://materializeinc.slack.com/archives/CL68GT3AT/p1689805092813149?thread_ts=1689797856.040739&cid=CL68GT3AT

Reply:
Happy to leave it as is, see my other comment